cluster/gce/list-resources.sh

```diff
@@ -49,7 +49,7 @@ function gcloud-compute-list() {
   while true; do
     if result=$(gcloud compute ${resource} list --project=${PROJECT} ${@:2}); then
       if [[ ! -z "${GREP_REGEX}" ]]; then
-        result=$(echo "${result}" | grep "${GREP_REGEX}")
+        result=$(echo "${result}" | grep "${GREP_REGEX}" || true)
```

Comment:
@madhusudancs, `gcloud-compute-list` function is just listing resources. listing resources is called before and after the tests, if there is any difference then there is resource leak. 
Here in the script since errexit is set, as soon as grep does not match pattern, the script exit with non-zero error code. showing 0 resource is also a valid case and we should just continue.
for eg. in the failed test ListResource Before: there are 0 zero firewall-rules matching the GREP_REGEX, that should be ok.
```
W1117 15:27:04.565] Project: k8s-jkns-e2e-gce-federation
W1117 15:27:04.566] Region: 
W1117 15:27:04.566] Zone: 
W1117 15:27:04.566] Instance prefix: bootstrap-e2e
W1117 15:27:04.566] Network: bootstrap-e2e
W1117 15:27:04.566] Provider: gce
W1117 15:27:04.566] 
W1117 15:27:04.567] 
W1117 15:27:04.567] [ instance-templates ]
W1117 15:27:04.567] 
W1117 15:27:04.567] 
W1117 15:27:04.567] 
W1117 15:27:04.567] [ instance-groups ]
W1117 15:27:04.568] 
W1117 15:27:04.568] 
W1117 15:27:04.568] 
W1117 15:27:04.568] [ instances ]
W1117 15:27:04.568] 
W1117 15:27:04.568] 
W1117 15:27:04.569] 
W1117 15:27:04.569] [ disks ]
W1117 15:27:04.569] 
W1117 15:27:04.569] 
W1117 15:27:04.569] 
W1117 15:27:04.569] [ addresses ]
W1117 15:27:04.569] NAME                              REGION       ADDRESS          STATUS
W1117 15:27:04.570] aac2c2791901211e6ad5242010a80000  us-central1  104.198.56.62    RESERVED
W1117 15:27:04.570] ac6321b9c80fe11e6ab6042010a80000  us-central1  104.197.171.242  RESERVED
W1117 15:27:04.570] 
W1117 15:27:04.570] 
W1117 15:27:04.570] [ routes ]
W1117 15:27:04.570] NAME                            NETWORK      DEST_RANGE     NEXT_HOP                  PRIORITY
W1117 15:27:04.570] 
W1117 15:27:04.570] 
W1117 15:27:04.570] [ firewall-rules ]
W1117 15:27:04.571]  error running Federated Ginkgo tests: exit status 1 Failed to list resources (exit status 1):
```